src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -878,9 +878,9 @@ def __flatten_key_value(key, value, flattened_data, depth, separator):
             if key in flattened_data:
                 logger.debug(
                     "The key %s already exist, value has been overwritten.", key)
-            flattened_data[key] = str(value)
+            flattened_data[key] = value if isinstance(value, str) else json.dumps(value)  # Ensure boolean values are properly stringified.
     else:
-        flattened_data[key] = str(value)
+        flattened_data[key] = value if isinstance(value, str) else json.dumps(value)
```

Comment:
Do we need to check `isinstance(value, str)`? Cant we always do `json.dumps(value)`?

Reply:
This is because string values will be represented with escaped quotes. e.g.,
For a json value indicated in a file as:
`{ "SomeString": "New value"  }`

The value will be exported to the configuration store with the quotes maintained. 
![image](https://user-images.githubusercontent.com/102393878/226058512-a00142cd-e80f-4199-93f6-680c3bd897fd.png)